flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1182,6 +1206,7 @@ def teardown_example(self, token):
         """
         Cleanup after running a hypothesis example.
         """
+        umount_all(self.deployer.mountroot)
```

Comment:
Unfortunately, `umount_all` returns a deferred, so
- this won't complete immediately
- it may end up cleaning up things from the next iteration.

The simplest solution(:frowning:) would be to change `umount_all` to not be asynchronous. A better long term solution would be to teach hypothesis how to run deferred running functions (by having an executor that does that[1]) but that is outside the scope of this issue.

[1] Maybe something that does `@hypothesis.executors.executor.extend(AyncTestCase)`.